README.md

```diff
@@ -129,12 +130,17 @@ Retriable.retriable(on: [Timeout::Error, Errno::ECONNRESET]) do
 end
 ```
 
-You can also specify a Hash of exceptions where the values are either `nil`, a single `Regexp` pattern, or an array of `Regexp`s.
+You can also specify a `Hash` where the keys are `Exception` subclasses and the values are either:
+
+A `Regexp` (or array of `Regexp`s). If any of the `Regexp`s match the exception's message, the block will be retried.
+A `Proc` (or array of `Proc`s) that evaluates the exception being handled and returns `true` if the block should be retried. If any of the procs in the list return `true`, the block will be retried.
+You can also mix and match `Proc`s and `Regexp`s in an `Array`
```

Comment:
this is all sort of duplicative of the docs [above](https://github.com/kamui/retriable#configuring-which-options-to-retry-with-on)... it's probably better to just link back there than to duplicate the explanation.

Reply:
I'm just following the style of this documentation. Above there are a little expanded version of what's on those bullet points as well.